rfcs/regex_feature.md

```diff
@@ -0,0 +1,89 @@
+# Regex feature
+
+A feature based on regex pattern matching that MindMeld's NER models can use to detect custom entities. 
+
+- Developers expect NER models to correctly infer that the query `AVET128-321` is a `pid` entity 
+without providing surrounding sentence context to the classifier. This is more relevant in services that
+just use MindMeld for NLP while doing their own dialog management. 
+- Developers also want granular distinctions between entities that occur in the same context, 
+for example, `look up {SKU120|sku}` and `look up {123-ABC-12|pid}`. Currently, we ask developers to defer
+these types of distinction to the business logic in the dialogue manager.
+- DialogFlow, NLTK and other similar platforms have the regex matching feature. Moreover, we already have an 
```

Comment:
How well does regex matching perform in those frameworks? We have concerns below about performance but if implementing this puts us on part with similar tools, I don't know if it's that worrisome.

On the other hand, if the performance gap is that broad, we may want to carefully consider how to implement this.

Reply:
We havent done a formal performance analysis between DF and MindMeld, we might want to plan on that in the future.